### ISSUE TITLE
Discussion of making build 1.8.0

### ISSUE THREAD

**Issue by liquidpele** [novus/nvd3#932]
@robinfhu @portante @joshjordan 

Before we start merging in any more features/etc, I think this may be a good time to cut 1.8.x out, which has several fixes and the new tooltip code.   Thoughts?

**Comment by robinfhu** [comment]
Thanks for opening this.

I haven't really had a chance to play around with nvd3 development yet.

I do remember this https://github.com/novus/nvd3/issues/895 being a somewhat annoying issue.  

But yeah we should release the 1.8 version.  I'll have some time Friday to test out nvd3's features.

**Comment by liquidpele** [comment]
Oh yea, #895 I should finish that PR, forgot about that.

**Comment by joshjordan** [comment]
The new tooltip code breaks backwards compatibility, yes? If we're using
semantic versioning (and I'd argue that we should be), we either need to
make the API backwards compatible, or bump the major version.

I'd advocate for the former, since I doubt tooltips is worth a major
version.

**Comment by liquidpele** [comment]
So, there are two aspects to the api changes....

1. the "tooltip" and "tooltipContent" options are deprecated... they're still there, and even try to use the new options on the tooltip object if called, and will print a warning in the js console.  
2. the data passed to the tooltip functions changed, so anyone trying to use a custom tooltip will probably see js errors due to their custom functions.  This really can't be made backward compatible because previously each chart was passing different stuff.

Really either the major number or minor number can be used for api changes, it's up to you guys really I guess.  

One option that I thought of was to have a special try/catch if they use the deprecated options so that if the tooltip errors out from the second issue above then it can log more helpful information, but that might be overkill since anyone writing their own custom tooltip can probably figure it out themselves...

**Comment by robinfhu** [comment]
Also need to remember to update the nvd3 website with new version, and ensure the examples work

**Comment by robinfhu** [comment]
Merged in #895, so that is taken care of.

**Comment by robinfhu** [comment]
Built nv.d3.js and updated the v1.8.1-alpha release tag.

**Comment by robinfhu** [comment]
We should also write some release notes in the Readme

**Comment by jjzazuet** [comment]
Also, please update the NPM registry whenever you guys get to release 1.8. Thanks! :)

**Comment by portante** [comment]
@liquidpele, @robinfhu, FWIW, I am reviewing the behaviors of 1.8.1-alpha with our code to see how it holds up.  I just encountered the voronoi half-edge problem again, which says something is not quite right, as I thought we fixed that.  I'll see what I can find.

**Comment by liquidpele** [comment]
The half-edge problem?

**Comment by portante** [comment]
See: https://github.com/mbostock/d3/pull/2238

**Comment by liquidpele** [comment]
You said we remove duplicates....  actually we don't, the code from https://github.com/novus/nvd3/pull/584 was merged into the 2.0 refactor, but that never went anywhere so it was never included into the current nvd3 versions.   I "fixed" it a while back by making the jitter it adds larger, but that's really all we've done around that...

**Comment by portante** [comment]
@liquidpele, well, yes, but I submitted code to do just that ... (looks)

**Comment by portante** [comment]
@liquidpele, oy, I have not gotten back to this!  https://github.com/novus/nvd3/pull/798

**Comment by jstrength** [comment]
Any ETA on this release?

**Comment by liquidpele** [comment]
Well, I want to fix https://github.com/novus/nvd3/issues/952 first, but other than that it's pretty much ready imho.

**Comment by jstrength** [comment]
Ok thanks, I sent a pull request that fixes that issue.

**Comment by liquidpele** [comment]
So is there anything else that needs to be done for this release @robinfhu @joshjordan @portante ?

**Comment by joshjordan** [comment]
Ship it.

**Comment by LeeMendelowitz** [comment]
I'm about to put in a PR that fixes interactive guidelines for line charts with missing data, if you guys wanna take a look before shipping

**Comment by LeeMendelowitz** [comment]
I just opened issue https://github.com/novus/nvd3/issues/963 and PR https://github.com/novus/nvd3/pull/962 which is my proposed fix. This bug was impacting me on a project I'm working on with discontinuous line charts.

**Comment by RenatoUtsch** [comment]
I think #810 (duplicates: #884, #937) is worth fixing before the release.

**Comment by portante** [comment]
@liquidpele, @robinfhu, @joshjordan, I don't have anything else, though I was hoping I'd have the time to get https://github.com/novus/nvd3/pull/798 finalized, since it is significant for us, but it will have to wait until a follow-on release. Thanks for considering, though.

**Comment by robinfhu** [comment]
Cool thanks!

**Comment by liquidpele** [comment]
I actually don't think @LeeMendelowitz's issue is worth holding it up for since it was pre-existing but if we can fit it in before hand then sure.   @RenatoUtsch's issue I think is way more serious since it'll affect more people, but I'm not sure what the "correct" behavior for it should be.

**Comment by RenatoUtsch** [comment]
This PR can be merged also (hehe): #967

**Comment by RenatoUtsch** [comment]
The following example pages are also rather buggy:
- examples/legend.html (legend being clipped)
- examples/pieChart.html (pie charts with wrong label positioning / missing part of the chart)
- examples/lineChart.html (referring to #963)
- examples/scatterChart.html (guideline broken)

Also, some examples doesn't exist anymore, the examples/index.html page needs to be updated (or removed).

**Comment by liquidpele** [comment]
the pie chart thing sounds like https://github.com/novus/nvd3/issues/861

#968 just got the fixed merged in. 

not sure about the scatter thing...

**Comment by RenatoUtsch** [comment]
Opened a separate issue for the scatter chart, it seems it is totally broken (not only the distribution - it was not the guideline, but the distribution): #971

**Comment by RenatoUtsch** [comment]
Another thing: a lot of the tests and examples use old tooltip code. The ones that give warnings about tooltips should be rewritten using the new tooltip.

**Comment by jjzazuet** [comment]
So I'm guessing there's still critical stuff to iron out right?

Thanks!

**Comment by liquidpele** [comment]
Yea, the only thing I'm concerned about is the other part of https://github.com/novus/nvd3/issues/971

**Comment by RenatoUtsch** [comment]
Yeah, any news on that?

**Comment by liquidpele** [comment]
built.
